Description: fix header needed to check OpenGL ES 2
Author: Dmitry Shachnev <mitya57@debian.org>
Bug-Upstream: https://github.com/annulen/webkit/issues/598
Last-Update: 2017-08-27

--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -1057,7 +1057,7 @@
 
 #if PLATFORM(QT)
 #ifdef __cplusplus
-#include <qglobal.h>
+#include <QtGui/qtguiglobal.h>
 #endif
 #if defined(QT_OPENGL_ES_2) && !defined(USE_OPENGL_ES_2)
 #define USE_OPENGL_ES_2 1
